handler/core/client/client.go

```diff
@@ -47,14 +47,14 @@ func (c *ClientCredentialsGrantHandler) HandleTokenEndpointRequest(ctx context.C
 		return nil
 	}
 
-	access, err := c.Enigma.GenerateChallenge(requester.GetClient().GetHashedSecret())
+	token, signature, err := c.AccessTokenStrategy.GenerateAccessToken(ctx, req, requester, session)
```

Comment:
instead of making an implicit assumption that every token is of type Challenge, we now only require that the token is verifyable with a signature that can be used for database lookup